modules/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVM.java

```diff
@@ -174,6 +184,9 @@ public void run(Context context) {
 
         try {
             exec();
+            if (context.isDebugEnabled()) {
+                VMDebugManager.getInstance().setDone(true);
```

Comment:
service path debug works, in that path we are not using this(done parameter) yet, so in service path debug, even you can have non blocking calls